src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -37,8 +37,9 @@ namespace detail
  * \param [in] quad the mfem integration rule containing nodes and weights
  * \return the value of the integral
  */
+template <typename T, int NDIMS>
 inline double evaluate_line_integral_component(
-  const primal::BezierCurve<double, 2>& c,
+  const primal::BezierCurve<T, NDIMS>& c,
   std::function<double(Point2D)> scalar_integrand,
```

Comment:
Should this still be a `Point2D`? Or does the type `T` and/or dimension `NDIMS` need to update as well?

Reply:
You're right, while the line integrals should work in arbitrary dimensions, I'd forgotten that the way we define the `std::function` integrands restricts them to 2D. I still feel as if it should work for arbitrary dimensions, or at least be templated instead of `double` precision, but I'll revert these changes and make a more thorough adjustment later if it comes to it.